opta/module_processors/aws_k8s_service.py

```diff
@@ -85,13 +87,19 @@ def process(self, module_idx: int) -> None:
         }
         if "image_tag" in self.layer.variables:
             self.module.data["tag"] = self.layer.variables["image_tag"]
+        seen = set()
```

Comment:
What are you trying to do here? Why would we have duplicates in `link_secrets`?

Reply:
It's a long story-- basically the process function get called multiple times with the same stateful module object and so this deduplicates secrets being set from the different calls

Reply:
Ouch that sounds like it will bite us down the line. We shouldn't maintain stateful objects.
But not a blocker for this PR!